### PR TITLE
riscv: syscall SYS_switch_context and SYS_restore_context use 0 para

### PR DESCRIPTION
**Pull request by hujun260**
*Note: Please adhere to [Contributing Guidelines](https://github.com/apache/nuttx/blob/master/CONTRIBUTING.md).*

## Summary
riscv: syscall SYS_switch_context and SYS_restore_context use 0 para
reason:
simplify context switch
sys_call0(SYS_switch_context)
sys_call0(SYS_restore_context)

size nuttx
before
   text    data     bss     dec     hex filename
 148021     921   26944  175886   2af0e nuttx

after
   text    data     bss     dec     hex filename
 147995     921   26928  175844   2aee4 nuttx

size reduce -42
## Impact
risc-v

## Testing

ci  ostest 
 rv-virt:smp

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



Yes, this PR appears to meet the basic NuttX requirements, although more detail would strengthen it.

Here's a breakdown of why and where it could be improved:

**Strengths:**

* **Clear Summary:** The summary explains the *what* and *why* of the change: simplifying context switching by reducing the parameters of two syscalls.  The size reduction is a nice quantifiable benefit.
* **Impact Section (Partially Addressed):**  It identifies the affected architecture (RISC-V).
* **Testing Information (Partially Addressed):** It mentions CI, `ostest`, and `rv-virt:smp`, indicating some testing was performed.

**Weaknesses & Areas for Improvement:**

* **Summary (More Detail):** *How* does removing the parameters simplify context switching?  Are these parameters unused? Were they previously passed unnecessarily?  This detail is crucial.
* **Impact Section (Incomplete):**
    * **User Impact:** Even if no user changes are needed, explicitly state "NO".
    * **Build Impact:**  Explicitly state "NO" or detail any changes.
    * **Hardware Impact:**  While RISC-V is mentioned, be more specific. Are all RISC-V boards affected? Any particular configurations?
    * **Documentation:** Does this change require documentation updates?  If not, state "NO".
    * **Security:**  Explicitly state "NO" or address any potential implications.
    * **Compatibility:** Explicitly state "NO" if no compatibility issues are expected.
* **Testing Section (Insufficient):**
    * **Build Host Details:** Provide details of your build host OS, CPU architecture, and compiler version.
    * **Target Details:** Be more specific about the RISC-V target.  "rv-virt:smp" is a start, but what specific configuration?  QEMU version?
    * **Missing Logs:** The "Testing logs before change" and "Testing logs after change" sections are empty.  Include relevant log snippets demonstrating the change's effect.  Show how the context switching behavior is different (ideally improved).  If size is a focus, showing size output before/after is excellent, but also demonstrate *functional* correctness.


**Example of Improved Testing Section:**

```
Testing

I confirm that changes are verified on local setup and works as intended:
* Build Host(s): Linux (Ubuntu 22.04), x86_64, GCC 11.3.0
* Target(s): RISC-V (QEMU 7.2.0), rv-virt:smp config

Testing logs before change:
```
$ size nuttx
   text    data     bss     dec     hex filename
 148021     921   26944  175886   2af0e nuttx
# Example:  Show some timing or profiling data related to context switching if applicable.
```

Testing logs after change:
```
$ size nuttx
   text    data     bss     dec     hex filename
 147995     921   26928  175844   2aee4 nuttx
# Example: Show corresponding data after the change.  Demonstrate improvement.
```

By addressing these weaknesses, the PR will be significantly stronger and more likely to be accepted.  Remember, providing clear and comprehensive information makes it easier for reviewers to understand and evaluate your contribution.

File: arch/risc-v/src/common/riscv_swint.c [hunk 1]
```diff
@@ -342,21 +310,17 @@ int riscv_swint(int irq, void *context, void *arg)
    * switch
    */
 
-  if (regs != new_regs)
-    {
-      restore_critical_section(this_task(), this_cpu());
-
 #ifdef CONFIG_DEBUG_SYSCALL_INFO
+  if (cmd > SYS_switch_context)
```
Comment (pussuw) [inline arch/risc-v/src/common/riscv_swint.c:314]: Why > ?

File: arch/risc-v/src/common/riscv_sigdeliver.c [hunk 2]
```diff
@@ -167,5 +167,5 @@ void riscv_sigdeliver(void)
   g_running_tasks[this_cpu()] = NULL;
```
Comment (pussuw) [inline arch/risc-v/src/common/riscv_sigdeliver.c:167]: This will now cause a crash, you must ensure this patch is merged only after the other patch which fixes this is merged.

Comment (hujun260) [reply arch/risc-v/src/common/riscv_sigdeliver.c:167]: It might cause conflicts, but it won't crash.

File: arch/risc-v/src/common/riscv_swint.c (hunk 1 above)
Comment (hujun260) [reply arch/risc-v/src/common/riscv_swint.c]: Updated

**Review by xiaoxiang781216 — APPROVED** [review]
